### PR TITLE
Update existing samples in order to use new chain pattern sugar

### PR DESCRIPTION
**Pull request by nasdan**
- Update typings
- Update sample 01 SignupForm

### CODE REVIEW

File: lib/lcformvalidation.d.ts [hunk 1]
```diff
@@ -31,8 +31,8 @@
       isFormPristine(): boolean;
       validateFullForm(vm: any): Promise<FormValidationResult>;
       triggerFieldValidation(vm: any, key: string, value: any, filter?: any): Promise<FieldValidationResult>;
-      addFieldValidation(key : string, validation : (vm, value) => FieldValidationResult, filter? : any);
-      addFieldValidationAsync(key : string, validation : (vm, value) => Promise<FieldValidationResult>, filter? : any);
+      addFieldValidation(key : string, validation : (vm, value) => FieldValidationResult, filter? : any): IValidationEngine;
+      addFieldValidationAsync(key : string, validation : (vm, value) => Promise<FieldValidationResult>, filter? : any): IValidationEngine;
```
Comment (brauliodiez) [inline lib/lcformvalidation.d.ts:35]: That what's something we missed on @crsanti review, he didn't updated the typescript definitions. 

Lesson learned from this case:

- Avoid extra noise, just focus on the case.
- Migrate samples in order to use new features.

**Comment by brauliodiez** [comment]
Good Stuff, @Nasdan can you propagate the changes to the other two samples?

Just use this branch and PR, once ready gimme a buzz I will perform a last review and procceed to merge.

Once we implement the multiple validation helper code will be lot cleaner.

**Comment by nasdan** [comment]
The other two samples only have one validation. That's why i didn't update it

File: lib/lcformvalidation.d.ts (hunk 1 above)
Comment (nasdan) [reply lib/lcformvalidation.d.ts:35]: - Add specs using this feature.

(I can add these specs in this PR if you want)

**Pull request merged**